astropy/coordinates/tests/test_sky_coord_velocities.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""
+Tests for putting velocity differentials into SkyCoord objects.
+"""
+
+import pytest
+
+import numpy as np
+
+from ... import units as u
+from ...tests.helper import assert_quantity_allclose
+from .. import (SkyCoord, ICRS, SphericalRepresentation, SphericalDifferential,
+                SphericalCosLatDifferential, Galactic, PrecessedGeocentric)
+
+try:
+    import scipy
+    HAS_SCIPY = True
+except ImportError:
+    HAS_SCIPY = False
+
+def test_creation_frameobjs():
+    i = ICRS(1*u.deg, 2*u.deg, pm_ra_cosdec=.2*u.mas/u.yr, pm_dec=.1*u.mas/u.yr)
+    sc = SkyCoord(i)
+
+    for attrnm in ['ra', 'dec', 'pm_ra_cosdec', 'pm_dec']:
+        assert_quantity_allclose(getattr(i, attrnm), getattr(sc, attrnm))
+
+    sc_nod = SkyCoord(ICRS(1*u.deg, 2*u.deg))
+
+    for attrnm in ['ra', 'dec']:
+        assert_quantity_allclose(getattr(sc, attrnm), getattr(sc_nod, attrnm))
+
+
+def test_creation_attrs():
+    sc1 = SkyCoord(1*u.deg, 2*u.deg,
+                   pm_ra_cosdec=.2*u.mas/u.yr, pm_dec=.1*u.mas/u.yr,
+                   frame='fk5')
+    assert_quantity_allclose(sc1.ra, 1*u.deg)
+    assert_quantity_allclose(sc1.dec, 2*u.deg)
+    assert_quantity_allclose(sc1.pm_ra_cosdec, .2*u.arcsec/u.kyr)
+    assert_quantity_allclose(sc1.pm_dec, .1*u.arcsec/u.kyr)
+
+    sc2 = SkyCoord(1*u.deg, 2*u.deg,
+                   pm_ra=.2*u.mas/u.yr, pm_dec=.1*u.mas/u.yr,
+                   differential_cls=SphericalDifferential)
+    assert_quantity_allclose(sc2.ra, 1*u.deg)
+    assert_quantity_allclose(sc2.dec, 2*u.deg)
+    assert_quantity_allclose(sc2.pm_ra, .2*u.arcsec/u.kyr)
+    assert_quantity_allclose(sc2.pm_dec, .1*u.arcsec/u.kyr)
+
+    sc3 = SkyCoord('1:2:3 4:5:6',
+                   pm_ra_cosdec=.2*u.mas/u.yr, pm_dec=.1*u.mas/u.yr,
+                   unit=(u.hour, u.deg))
+
+    assert_quantity_allclose(sc3.ra, 1*u.hourangle + 2*u.arcmin*15 + 3*u.arcsec*15)
+    assert_quantity_allclose(sc3.dec, 4*u.deg + 5*u.arcmin + 6*u.arcsec)
+    # might as well check with sillier units?
+    assert_quantity_allclose(sc3.pm_ra_cosdec, 1.2776637006616473e-07 * u.arcmin / u.fortnight)
+    assert_quantity_allclose(sc3.pm_dec, 6.388318503308237e-08 * u.arcmin / u.fortnight)
+
+@pytest.mark.xfail  #TODO: FIX
+def test_creation_copy():
+    i = ICRS(1*u.deg, 2*u.deg, pm_ra_cosdec=.2*u.mas/u.yr, pm_dec=.1*u.mas/u.yr)
```

Comment:
Since you're using the same inputs, I would suggest making a test class, in which these are set up in `setup`.